custom_components/hikvision_next/isapi.py

```diff
@@ -40,309 +41,518 @@
+            if not notifications:
+                continue
+
+            notifications = notifications.get("EventTriggerNotification", [])
+
+            if not isinstance(notifications, list):
+                notifications = [notifications]
+
+            # Translate to alternate IDs
+            if event_type.lower() in EVENTS_ALTERNATE_ID.keys():
+                event_type = EVENTS_ALTERNATE_ID[event_type.lower()]
+
+            events.append(
+                SupportedEventsInfo(
+                    channel_id=int(channel),
+                    event_id=event_type.lower(),
+                    notifications=[
+                        notify.get("notificationMethod") for notify in notifications
+                    ]
+                    if notifications
+                    else [],
+                )
             )
 
-    async def _get_camera_events_capabilities(self) -> list[str]:
-        """Get available camera events."""
+        return events
 
-        data = await self.isapi.Event.capabilities(method=GET)
-        _LOGGER.debug("%s/ISAPI/Event/capabilities %s", self.isapi.host, data)
-        basic_events = [
-            event_id.lower().replace("issupport", "")
-            for event_id, is_supported in data["EventCap"].items()
-            if str_to_bool(is_supported)
-        ]
+    def get_event_url(
+        self, event_id: str, channel_id: int, is_nvr: bool, camera_type: str
+    ) -> str:
+        """Get event ISAPI URL."""
 
-        data = await self.isapi.Smart.capabilities(method=GET)
-        _LOGGER.debug("%s/ISAPI/Smart/capabilities %s", self.isapi.host, data)
-        smart_events = [
-            event_id.lower().replace("issupport", "")
-            for event_id, is_supported in data["SmartCap"].items()
-            if str_to_bool(is_supported)
-        ]
+        event_type = EVENTS[event_id]["type"]
+        slug = EVENTS[event_id]["slug"]
 
-        return basic_events + smart_events
+        if (
+            is_nvr
+            and camera_type == DEVICE_TYPE_IP_CAMERA
+            and event_type == EVENT_BASIC
+        ):
+            # ISAPI/ContentMgmt/InputProxy/channels/{channel_id}/video/{event}
+            url = f"ContentMgmt/InputProxy/channels/{channel_id}/video/{slug}"
+        elif (
+            is_nvr and camera_type == DEVICE_TYPE_ANALOG_CAMERA or not is_nvr
+        ) and event_type == EVENT_BASIC:
+            # ISAPI/System/Video/inputs/channels/{channel_id}/{event}
+            url = f"System/Video/inputs/channels/{channel_id}/{slug}"
+        else:
+            # ISAPI/Smart/{event}/{channel_id}
+            url = f"Smart/{slug}/{channel_id}"
+        return url
 
-    async def _get_channel_events_capabilities(self, channel_id: str) -> list[str]:
-        """Get available NVR channel events."""
+    async def get_camera_streams(self, channel_id: int) -> list[CameraStreamInfo]:
+        # TODO - Improve efficiency of this by reading once and adding to each camera
```

Comment:
Do you have ISAPI documentation? If not I could share. Maybe there we will find better way?